sidx.go

```diff
@@ -0,0 +1,39 @@
+package mp4
+
+import "fmt"
+
+func BoxTypeSidx() BoxType { return StrToBoxType("sidx") }
+
+func init() {
+	AddBoxDef(&Sidx{}, 0)
+}
+
+// Sidx is ISOBMFF Sidx box type
+type Sidx struct {
+	FullBox    `mp4:"extend"`
+	ReferenceID uint32 `mp4:"size=32"`
+	TimeScale uint32 `mp4:"size=32"`
+	EarliestPresentationTime uint32 `mp4:"size=32"`
+	FirstOffset uint32 `mp4:"size=32"`
```

Comment:
If `Sidx.Version` equals to 1, `Sidx.EarliestPresentationTime` and `Sidx.FirstOffset` have 64 bits.

So could you use `ver` option as follows?

```
	EarliestPresentationTime uint32 `mp4:"size=32,ver=0"`
	FirstOffset              uint32 `mp4:"size=32,ver=0"`
	EarliestPresentationTime uint64 `mp4:"size=64,ver=1"`
	FirstOffset              uint64 `mp4:"size=64,ver=1"`
```

Reply:
https://github.com/abema/go-mp4/pull/18/files#diff-6322880a133c5e374279625376187a10R8
Sorry, I was mistaken.
This PR accept only version 0 at line 8, and is correct.

Reply:
FYI, If supporting both of version 0 and version 1, you can write as follows:

```
func init() {
	AddBoxDef(&Sidx{}, 0, 1)
}

       :
       :
	EarliestPresentationTimeV0 uint32 `mp4:"size=32,ver=0"`
	FirstOffsetV0              uint32 `mp4:"size=32,ver=0"`
	EarliestPresentationTimeV1 uint64 `mp4:"size=64,nver=0"`
	FirstOffsetV1              uint64 `mp4:"size=64,nver=0"`
```